File: vripper-server/src/main/java/tn/mnlr/vripper/jpa/domain/Post.java
package tn.mnlr.vripper.jpa.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import tn.mnlr.vripper.jpa.domain.enums.Status;

import java.time.LocalDateTime;
import java.util.Collections;
import java.util.Objects;
import java.util.Set;

@Getter
@Setter
@ToString
@NoArgsConstructor
public class Post {

  @JsonIgnore private Long id;

  private Status status;

  private String postId;

  private String threadTitle;

  private String threadId;

  private String title;

  private String url;

  private int done;

  private int total;

  private Set<String> hosts;

  private String forum;

  @JsonIgnore private String securityToken;

  @JsonIgnore private String downloadDirectory;

  private boolean thanked;

  private Set<String> previews = Collections.emptySet();

  private Metadata metadata;

  private LocalDateTime addedOn;

  public Post(
      String title,
      String url,
      String postId,
      String threadId,
      String threadTitle,
      String forum,
      String securityToken) {
    this.title = title;
    this.url = url;
    this.postId = postId;
    this.forum = forum;
    this.threadId = threadId;
    this.threadTitle = threadTitle;
    this.securityToken = securityToken;
    status = Status.STOPPED;
    addedOn = LocalDateTime.now();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Post post = (Post) o;
    return Objects.equals(postId, post.postId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(postId);
  }
}

File: vripper-core/src/main/kotlin/me/mnlr/vripper/entities/Post.kt
package me.mnlr.vripper.entities

import com.fasterxml.jackson.annotation.JsonFormat
import me.mnlr.vripper.entities.domain.Status
import java.time.LocalDateTime

data class Post(
    var id: Long = -1,
    val postTitle: String,
    val threadTitle: String,
    val forum: String,
    val url: String,
    val token: String,
    val postId: String,
    val threadId: String,
    val total: Int,
    val hosts: Set<String>,
    val downloadDirectory: String,
    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss") val addedOn: LocalDateTime = LocalDateTime.now(),
    var status: Status = Status.STOPPED,
    var done: Int = 0,
    var rank: Int = Int.MAX_VALUE
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Post

        if (postId != other.postId) return false

        return true
    }

    override fun hashCode(): Int {
        return postId.hashCode()
    }
}